src/main/scala/com/microsoft/hyperspace/util/DataAnalysisUtils.scala

```diff
@@ -0,0 +1,278 @@
+
+    val valueStr = values.mkString(",")
+    val yTickNum = 10.min(yMax)
+
+    s"""
+    <!DOCTYPE html>
+    <meta charset="utf-8">
+    <script src="https://d3js.org/d3.v4.js"></script>
+    <div id="my_dataviz"></div>
+
+    <script>
+    var margin = {"top": 20, "right": 30, "bottom": 25, "left": 50 }
+    var width = 500;
+    var height = 500;
+    var rectWidth = ${500 / values.length};
+
+    var data = [$valueStr];
+
+    var xMax = ${values.length} * rectWidth;
+    var xScale = d3.scaleLinear()
+      .domain([0, xMax])
+      .range([margin.left, width - margin.right]);
+
+    var yMax = $yMax;
+    var yScale = d3.scaleLinear()
+      .domain([0, yMax])
+      .range([height - margin.bottom, margin.top]);
+
+    var svg = d3.select('svg');
+
+    var rect = svg.selectAll('rect')
+      .data(data)
+      .enter().append('rect')
+      .attr('x', function(d, i) {
+        return xScale(i * rectWidth)})
+      .attr('y', function(d) {
+        return yScale(d)})
+      .attr('width', xScale(rectWidth) - margin.left)
+      .attr('height', function(d) {
+        return height - margin.bottom - yScale(d)})
+      .attr('fill', "#AED6F1")
+      .attr('margin', 0);
+
+    var xAxis = d3.axisBottom()
+      .scale(xScale)
+      .tickFormat((d, i) => ['Min value', 'Max value'][i])
+      .ticks(1);
+    var yAxis = d3.axisLeft()
+      .scale(yScale)
+      .tickFormat(d3.format('d'))
+      .ticks($yTickNum);
+
+    svg.append('g')
+      .attr('transform', 'translate(' + [0, height - margin.bottom] + ')')
+      .call(xAxis);
+    svg.append('g')
+      .attr('transform', 'translate(' + [margin.left, 0] + ')')
+      .call(yAxis);
+
+    svg.append("text")
+      .attr("text-anchor", "middle")
+      .attr("transform", "translate(" + 15 + "," + (height / 2) + ")rotate(-90)")
+      .style("font-size", "15px")
+      .text("Max. number of files to lookup");
+
+    svg.append("text")
+      .attr("text-anchor", "middle")
+      .attr("transform", "translate(" + (width / 2) + "," + (height - 4) + ")")
+      .style("font-size", "15px")
+      .text("Column value range");
+
+    </script>
+    <body><svg width=500 height=500></svg></body>
+    """
+  }
+
+  def minMaxAnalysis(df: DataFrame, colName: String, htmlFormat: Boolean = false): String = {
+    val dataType = df.schema.fields.find(_.name.equals(colName)).get.dataType
+    val numeric = try {
+      TypeUtils.getNumeric(dataType)
+    } catch {
+      case _: Exception =>
+        throw HyperspaceException(
+          s"minMaxAnalysis only supports Numeric type. colName=$colName, dataType=$dataType")
+    }
+
+    val distFigureLen = if (htmlFormat) 20 else 50
+    val newline = if (htmlFormat) "</br>" else System.lineSeparator()
+    val res = df.inputFiles.par.map { f =>
+      val file = df.sparkSession.read.parquet(f)
+      f -> file.agg(min(col(colName)), max(col(colName))).collect.toSeq.headOption
+    }.toIndexedSeq.filter(_._2.isDefined)
```

Comment:
This is interesting. Currently I'm using `groupBy(input_file_name()).agg(...)` for data skipping indexes but also considering this approach, but with Spark RDD to parallelize. Did you consider using `spark.sparkContext.parallelize` instead of Scala's `.par`? My understanding is that if you use `.par` then Spark can't efficiently schedule jobs because the parallelization is opaque to Spark.

Reply:
current ver - ".par.map" triggers a new spark job for each file with multiple threads. It can be done by multiple tasks if it's a large file. So parallelism is not an issue for this; actually this is faster than groupBy impl. (though not sure why groupBy is slower; maybe it's just because "input_file_name()" function ..?)
One problem is that the impl incurs numerous spark jobs if there're many files which can cause some complexity of spark history ui or other analysis tools. Considering the fact, I would prefer groupBy impl. though it's bit slower than this.

For `spark.sparkContext.parallize`, I'm not sure how I can use it instead of `.par`

Reply:
`spark.sparkContext.parallelize(df.inputFiles)` gives an RDD than you can work on. It is better than Scala's built-in parallel collections because Spark can efficiently schedule jobs. For Parquet files, you can directly read min/max stats with `ParquetFileReader` without reading the whole files.

```
def collectMinMax(path: String, colName: String): (Any, Any) = {
   // Read min/max using ParquetFileReader
}
spark.sparkContext.parallelize(df.inputFiles).map(collectMinMax(_, colName))
```

Unfortunately, I couldn't find yet how you can populate DF in the map function. The map function is executed in executor nodes so obviously `spark.read` doesn't work. But even if it only works for Parquet files, I think it's worth a try since the speed up would be huge and Parquet is a very popular format. This can be done for other formats supporting statistics such as ORC.

Reply:
I think we can use groupBy / min&max aggregation for now, as it can support most type of data sources. We can optimize the function for delta lake, but for hyperspace, let's keep the current impl. for the initial version and extend the feature later on demand.